src-built-in/adapters/indexedDBAdapter.js

```diff
@@ -0,0 +1,259 @@
+			let data = [];
+			if ( event.target.result && event.target.result.value) {
+				data = event.target.result;
+			}
+			Logger.system.debug("IndexedDBAdapter.keys for keyPreface=" + keyPreface + " keys=", data);
+			console.debug("IndexedDBAdapter.get keys keyPreface=" + keyPreface + " keys=", data);
+			cb(null, data);
+		};
+	};
+
+	/**
+	 * Delete method.
+	 * @param {object} params
+	 * @param {string} params.topic A topic under which the data should be stored.
+	 * @param {string} params.key The key whose value is being deleted.
+	 * @param {function} cb callback to be invoked upon completion
+	 */
+	this.delete = (params, cb) => {
+		const combinedKey = this.getCombinedKey(this, params);
+		Logger.system.debug("IndexedDBAdapter.delete for key=" + combinedKey);
+		console.debug("IndexedDBAdapter.delete for key=" + combinedKey);
+
+		var objectStore = db.transaction(["fsbl"], "readwrite").objectStore("fsbl");
+		var request = objectStore.delete(combinedKey);
+		request.onerror = function(event) {
+			Logger.system.error("IndexedDBAdapter.delete key=" + combinedKey + ", Error", event);
+			console.error("IndexedDBAdapter.delete key=" + combinedKey + ", Error", event);
+			return cb(event, { status: "failed" });
+		};
+		request.onsuccess = function(event) {
+			cb(null, { status: "success" });
+		};
+	};
+
+	/**
+	 * This method should be used very, very judiciously. It's essentially a method designed to wipe the database for a 
+	 * particular user.
+	 */
+	this.clearCache = (params, cb) => {
+		const userPreface = this.getUserPreface(this);
+		Logger.system.debug("IndexedDBAdapter.clearCache for userPreface=" + userPreface);
+		console.debug("IndexedDBAdapter.clearCache for userPreface=" + userPreface);
+				
+		let keyRange = IDBKeyRange.forPrefix(userPreface);
+		var objectStore = db.transaction(["fsbl"], "readwrite").objectStore("fsbl");
+		var request = objectStore.delete(keyRange);
+		request.onerror = function(event) {
+			Logger.system.debug("IndexedDBAdapter.clearCache failed Error", event, "userPreface=" + userPreface);
+			console.debug("IndexedDBAdapter.clearCache failed Error", event, "userPreface=" + userPreface);
+			return cb(event, { status: "failed" });
+		};
+		request.onsuccess = function(event) {
+			cb();
+		};
+	};
+
+	/**
+	 * Wipes the storage container.
+	 * @param {function} cb
+	 */
+	this.empty = (cb) => {
+		Logger.system.debug("IndexedDBAdapter.empty");
+		console.debug("IndexedDBAdapter.empty");
+		var objectStore = db.transaction(["fsbl"], "readwrite").objectStore("fsbl");
+		var request = objectStore.clear();
+		request.onerror = function(event) {
+			Logger.system.debug("IndexedDBAdapter.empty failed Error", event);
+			console.debug("IndexedDBAdapter.empty failed Error", event);
+			return cb(event, { status: "failed" });
+		};
+		request.onsuccess = function(event) {
+			cb();
+		};
+	};
+};
+
+IndexedDBAdapter.prototype = new BaseStorage();
+new IndexedDBAdapter("IndexedDBAdapter");
```

Comment:
@thorsent The fact that this script instantiates the IndexDBAdapter and does nothing seems very weird to me. I would have thought it would export the instance like the services does, or, even better, return the constructor so it can be instantiated by Finsemble. Either way, I think this could be made clearer during refactoring.

Reply:
Agreed.